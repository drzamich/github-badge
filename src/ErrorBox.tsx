import React from 'react';

interface ErrorBoxProps {

}

export const ErrorBox: React.FC<ErrorBoxProps> = () => null;
