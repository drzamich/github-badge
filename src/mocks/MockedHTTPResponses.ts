import { AxiosError, AxiosResponse } from 'axios';
import { user as ValidUserResponse, repos as ValidReposResponse } from './ValidAPIResponses.json';
import { Repository, User } from '../interfaces';

const defResponse = {
  headers: [],
  config: {},
};

const defSuccessResponse = {
  ...defResponse,
  status: 200,
  statusText: 'success',
};

const defErrorResponse: AxiosResponse = {
  ...defResponse,
  status: 500,
  statusText: 'success',
  data: null,
};

const defError: AxiosError = {
  config: {},
  isAxiosError: true,
  toJSON: () => ({}),
  name: '',
  message: '',
};

export const userReponse: AxiosResponse<User> = {
  data: ValidUserResponse,
  ...defSuccessResponse,
};

export const reposResponse: AxiosResponse<Repository[]> = {
  data: ValidReposResponse,
  ...defSuccessResponse,
};

export const networkError: AxiosError = {
  ...defError,
  request: {},
};

export const internalServerError: AxiosError = {
  ...defError,
  response: {
    ...defErrorResponse,
  },
};

export const notFoundError: AxiosError = {
  ...defError,
  response: {
    ...defErrorResponse,
    status: 404,
  },
};
