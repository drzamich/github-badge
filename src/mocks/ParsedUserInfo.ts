/* eslint-disable max-len */
import { User, Repository } from '../interfaces';

export const parsedUser: User = {
  name: 'Tejas Kumar',
  login: 'TejasQ',
  avatar_url: 'https://avatars1.githubusercontent.com/u/9947422?v=4',
  bio: 'I write code and love life.',
};

export const parsedRepos: Repository[] = [
  { full_name: 'TejasQ/add-gitignore', html_url: 'https://github.com/TejasQ/add-gitignore', stargazers_count: 450 },
  { full_name: 'TejasQ/anna-artemov.now.sh', html_url: 'https://github.com/TejasQ/anna-artemov.now.sh', stargazers_count: 5 },
  { full_name: 'TejasQ/awesome-cli-apps', html_url: 'https://github.com/TejasQ/awesome-cli-apps', stargazers_count: 3 },
];

export const finalUser: User = {
  ...parsedUser,
  repos: parsedRepos,
};
