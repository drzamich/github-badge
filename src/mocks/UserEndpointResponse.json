{
  "login": "TejasQ",
  "id": 9947422,
  "node_id": "MDQ6VXNlcjk5NDc0MjI=",
  "avatar_url": "https://avatars1.githubusercontent.com/u/9947422?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/TejasQ",
  "html_url": "https://github.com/TejasQ",
  "followers_url": "https://api.github.com/users/TejasQ/followers",
  "following_url": "https://api.github.com/users/TejasQ/following{/other_user}",
  "gists_url": "https://api.github.com/users/TejasQ/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/TejasQ/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/TejasQ/subscriptions",
  "organizations_url": "https://api.github.com/users/TejasQ/orgs",
  "repos_url": "https://api.github.com/users/TejasQ/repos",
  "events_url": "https://api.github.com/users/TejasQ/events{/privacy}",
  "received_events_url": "https://api.github.com/users/TejasQ/received_events",
  "type": "User",
  "site_admin": false,
  "name": "Tejas Kumar",
  "company": "@g2i",
  "blog": "https://twitter.com/tejaskumar_",
  "location": "Berlin, DE",
  "email": null,
  "hireable": true,
  "bio": "I write code and love life.",
  "twitter_username": null,
  "public_repos": 60,
  "public_gists": 1,
  "followers": 753,
  "following": 280,
  "created_at": "2014-11-25T13:12:53Z",
  "updated_at": "2020-11-25T21:05:04Z"
}