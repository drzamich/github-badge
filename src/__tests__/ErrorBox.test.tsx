import React from 'react';
import { render } from '@testing-library/react';
import { AxiosError } from 'axios';
import { internalServerError, networkError, notFoundError } from '../mocks/MockedHTTPResponses';
import { ErrorBox } from '../ErrorBox';

describe('ErrorBox', () => {
  const errors: [string, AxiosError][] = [
    ['internal server error', internalServerError],
    ['not found error', notFoundError],
    ['network error', networkError],
  ];

  it.each(errors)('Displays proper error message for a %s', (errorType, error) => {
    // Given
    // When
    const component = render(<ErrorBox error={error} />);
    // Then
    expect(component).toMatchSnapshot();
  });
});
