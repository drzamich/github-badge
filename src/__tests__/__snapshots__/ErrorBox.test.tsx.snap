// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox Displays proper error message for a internal server error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="sc-bdfBwQ fVVSDW"
      >
        Error occured. Please try again.
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      class="sc-bdfBwQ fVVSDW"
    >
      Error occured. Please try again.
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ErrorBox Displays proper error message for a network error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="sc-bdfBwQ fVVSDW"
      >
        Network issue. Please try again.
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      class="sc-bdfBwQ fVVSDW"
    >
      Network issue. Please try again.
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ErrorBox Displays proper error message for a not found error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="sc-bdfBwQ fVVSDW"
      >
        User not found. Please try again.
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      class="sc-bdfBwQ fVVSDW"
    >
      User not found. Please try again.
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
